### PR TITLE
Revert "Delete @types/gulp & @types/minimatch on posinstall"

### PR DESCRIPTION
**Pull request by coroiu**
Reverts bitwarden/clients#11509

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -212,7 +212,11 @@
     "@storybook/angular": {
       "zone.js": "$zone.js"
     },
-    "replacestream": "4.0.3"
+    "replacestream": "4.0.3",
+    "@types/minimatch": "3.0.5",
+    "@electron/asar": {
+      "@types/glob": "7.1.3"
+    }
```
Comment (Hinton) [inline package.json:219]: I don't think we need this any longer?

Comment (coroiu) [reply package.json:219]: I have no idea what any of this does, I just know minimatch is blocking lock-file maintenance, do you think we can remove minimatch entirely?

**Comment by Hinton** [comment]
@coroiu can you provide more context for this? We need to upgrade electron asar to the new version which inlines @types/gulp. Unfortunately they still depend on the old version of minimatch which probably causes conflicts with gulp.

**Comment by coroiu** [comment]
@Hinton https://github.com/bitwarden/clients/pull/11509#issuecomment-2420285523 says

> "This can be reverted once https://github.com/electron/asar/issues/330 is resolved."

And since that issue seems to be closed I opened this PR. I was hoping you had the context 😅

**Comment by Hinton** [comment]
Yes, once we upgrade asar the types/gulp portion should be fixed, not sure about minimatch. But we first have to ensure we get the new asar package in, which means updating the lockfile.

File: package.json (hunk 1 above)
Comment (Hinton) [inline package.json:219]: This should work now, both type dependencies were gone from my lockfile.
```suggestion
    "replacestream": "4.0.3"
```

**Comment by coroiu** [comment]
Closing since no longer needed

**Pull request closed without merging**